frostsnap_coordinator/src/signing.rs

```diff
@@ -84,8 +86,10 @@ impl SigningDispatcher {
 
     pub fn resend_sign_request(&mut self) -> Option<CoordinatorSendMessage> {
         if !self.need_to_send_to.is_empty() {
+            let target_destinations = Destination::from(self.need_to_send_to.clone());
+            self.need_to_send_to = BTreeSet::new();
             return Some(CoordinatorSendMessage {
-                target_destinations: Destination::from(self.need_to_send_to.drain()),
```

Comment:
I have swapped the the HashSet types to BTreeSet to better align with `frostsnap_core`. This drain was the only significant HashSet feature.

Reply:
Changing to BTreeSet is fine. I think you want to use `core::mem::take` here rather than `clone` and then set it to `new`.